src/lib/utils/thorchain/hooks/useThorchainFromAddress.tsx

```diff
@@ -0,0 +1,51 @@
+import type { AccountId, AssetId } from '@shapeshiftoss/caip'
+import type { HDWallet } from '@shapeshiftoss/hdwallet-core'
+import type { AccountMetadata } from '@shapeshiftoss/types'
+import { useQuery } from '@tanstack/react-query'
+import { reactQueries } from 'react-queries'
+import type { getThorchainLpPosition } from 'pages/ThorChainLP/queries/queries'
+import type { getThorchainSaversPosition } from 'state/slices/opportunitiesSlice/resolvers/thorchainsavers/utils'
+
+import type { getThorchainLendingPosition } from '../lending'
+
+type UseThorchainFromAddressArgs = {
+  accountId: AccountId | undefined
+  assetId: AssetId | undefined
+  opportunityId?: string | undefined
+  wallet: HDWallet | null
+  accountMetadata: AccountMetadata | undefined
+  getPosition:
+    | typeof getThorchainLendingPosition
+    | typeof getThorchainSaversPosition
+    | typeof getThorchainLpPosition
+  select?: (maybeAddress: string) => string | undefined
```

Comment:
whereas this one can be optional since a selector is consumer specific and not relevant to the hook itself meaning it can safely be omitted if desired
```suggestion
  select?: (maybeAddress: string) => string
```

Reply:
disregard, misread this being the function OR undefined, not the return type being string or undefined